reconcile/dashdotdb_dora.py

```diff
@@ -0,0 +1,480 @@
+        ] = threaded.run(
+            func=self.get_repo_changes,
+            iterable=repo_changes_iterable,
+            thread_pool_size=self.thread_pool_size,
+        )
+
+        # filter invalid ones
+        saastarget_dep_repo_changes = [
+            (st, dep, rc)
+            for (st, dep, rc) in saastarget_dep_repo_changes
+            if rc.repo_url and rc.ref_from and rc.ref_to and rc.ref_from != rc.ref_to
+        ]
+
+        unique_repo_changes = {rc for (_, _, rc) in saastarget_dep_repo_changes}
+
+        rc_commits_list = threaded.run(
+            func=self.compare,
+            iterable=unique_repo_changes,
+            thread_pool_size=self.thread_pool_size,
+        )
+
+        rc_commits = dict(rc_commits_list)
+
+        # we want to sort by oldest first, so that if the process crashes
+        # the call to latest_deployment from Dash.DB will return the appropriate
+        # value so that it is possible to continue
+        def sortkey(item: tuple[SaasTarget, Deployment, RepoChanges]) -> datetime:
+            (_, dep, _) = item
+            return dep.deployment_time
+
+        deployments = []
+        for st, dep, rc in sorted(saastarget_dep_repo_changes, key=sortkey):
+            deployments.append(self.build_deployment_post(st, dep, rc_commits[rc]))
+
+        if deployments:
+            self.post({"deployments": deployments})
+
+    def get_saastargets(self) -> list[SaasTarget]:
+        targets = []
+        for saas_file in get_saas_files():
+            if not saas_file.labels:
+                continue
+
+            if "dora" not in saas_file.labels:
+                continue
+
+            dora_target_env_names = saas_file.labels["dora"].split(",")
+            for rt in saas_file.resource_templates:
+                for target in rt.targets:
+                    ns = target.namespace
+                    target_env_name = ns.environment.name
+                    if target_env_name not in dora_target_env_names:
+                        continue
+
+                    pipeline = f"{rt.name}-{ns.cluster.name}-{ns.name}"
+                    saas_file_path = "data/" + saas_file.path.lstrip("/")
+                    saastarget = SaasTarget(
+                        app_name=saas_file.name,
+                        env_name=target_env_name,
+                        path=saas_file_path,
+                        resource_template=rt.name,
+                        namespace=ns.path,
+                        pipeline=pipeline,
+                    )
+
+                    targets.append(saastarget)
+
+                    LOG.info("marking SaaS file for processing: %s", saastarget)
+        return targets
+
+    def get_app_env_deployments(
+        self, app_env: AppEnv
+    ) -> tuple[AppEnv, list[Deployment]]:
+        since = self.get_latest(app_env)
+
+        if not since:
+            # grab last 90 days
+            since = self.start_timestamp - timedelta(days=90)
```

Comment:
reading deployments from the last 90 days means that deployments are reprocessed for quite some time. what was the thought process for this value?

Reply:
sry i misunderstood. this is basically just for an initial import and 90d is for sure fine with that